packages/catalog-generator/pom.xml

```diff
@@ -160,6 +160,26 @@
           </classPatterns>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>${version.maven-enforcer-plugin}</version>
+        <executions>
+          <execution>
+            <id>enforce-java-version</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireJavaVersion>
+                  <version>[17.0.0,17.0.999],[21.0.0,21.0.999]</version>
```

Comment:
This will prevent users with Java 22+ to build. or even 21.1
I do no tthink that it is what we want

Reply:
By reading version range rule defintion https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html, it seems like we are forbiding Java 17.1, 18, 19, 20 too

Reply:
I think we should go with only `17`

Reply:
Mmm, didn't like it
```
[ERROR] Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion failed with message:
[ERROR] Detected JDK version 17.0.12 (JAVA_HOME=/home/rmartinez/.sdkman/candidates/java/17.0.12-tem) is not in the allowed range [17,17].
```

Reply:
I think it should be `[1.0,)`

Reply:
no, either
* 17
* [17,)

Reply:
strange, just 17 is working locally for me:
![image](https://github.com/user-attachments/assets/34d328ba-a895-4af3-9681-f92b616194bf)

Reply:
:confused:

Reply:
I think the problem was `[17]` getting extrapolated to `[17,17]`. I missed the `17` alone.

Reply:
That was it, using `17` alone works with both `17` and `21`